EMBRACE_Client/ServerCommunicationController.m

```diff
@@ -0,0 +1,1265 @@
+    DDXMLElement *nodeUserActionID = [DDXMLElement elementWithName:@"User Action ID" stringValue:[NSString stringWithFormat:@"%ld",(long)UserActionIDTag]];
+    
+    //logging selection
+    DDXMLElement *nodeSelection = [DDXMLElement elementWithName:@"Selection" stringValue:@"Audio"];
+    
+    //Input parent node
+    DDXMLElement *nodeInput = [DDXMLElement elementWithName:@"Input"];
+    
+    //creating inpud children nodes
+    DDXMLElement *nodeAudioFilename = [DDXMLElement elementWithName:@"Audio Filename" stringValue:audioFileName];
+    DDXMLElement *nodeAudioLanguage = [DDXMLElement elementWithName:@"Audio Language" stringValue:LanguageType];
+    
+    //adding child nodes to Input parent
+    [nodeInput addChild:nodeAudioFilename];
+    [nodeInput addChild:nodeAudioLanguage];
+    
+    //logging action
+    DDXMLElement *nodeAction = [DDXMLElement elementWithName:@"Action" stringValue:computerAction];
+    
+    //logging Context
+    DDXMLElement *nodeContext = [[ServerCommunicationController sharedManager] returnContext:storyValue :chapterValue :pageValue :sentenceValue :stepValue];
+    
+    //add SAIC to UserAction parent
+    [nodeComputerAction addChild:nodeUserActionID];
+    [nodeComputerAction addChild:nodeSelection];
+    [nodeComputerAction addChild:nodeAction];
+    [nodeComputerAction addChild:nodeInput];
+    [nodeComputerAction addChild:nodeContext];
+    
+    //bool successfulWrite = [[ServerCommunicationController sharedManager] writeToFile:userNameString ofType:@"txt"];
+}
+
+/*
+ UserActionIDTag: current useractionID
+ Selection: ??
+ Action: Display Menu Items
+ Input: displayedMenuItems
+ Context: story, chapter, page, sentence, step, username, condition, experimenter
+ */
+-(void) logComputerDisplayMenuItems :  (NSArray *) displayedMenuInteractions :(NSArray *)displayedMenuImages : (NSArray*) displayedMenuRelationships : (NSString *) storyValue : (NSString *) chapterValue : (NSString *) pageValue : (NSString *) sentenceValue : (NSString *) stepValue
+{
+    UserActionIDTag++;
+    
+    //logging structure for user actions
+    DDXMLElement *nodeComputerAction = [DDXMLElement elementWithName:@"Computer Action"];
+    [study addChild:nodeComputerAction];
+    
+    //logging userAction relationship
+    DDXMLElement *nodeUserActionID = [DDXMLElement elementWithName:@"User Action ID" stringValue:[NSString stringWithFormat:@"%ld",(long)UserActionIDTag]];
+    
+    //logging selection
+    DDXMLElement *nodeSelection = [DDXMLElement elementWithName:@"Selection" stringValue:@"Image"];
+    
+    //logging Input
+    DDXMLElement *nodeInput = [DDXMLElement elementWithName:@"Input"];
+    DDXMLElement *nodeDisplayedMenuItem1;
+    DDXMLElement *nodeDisplayedMenuItem2;
+    DDXMLElement *nodeDisplayedMenuItem3;
+    
+    // checks the number of menu items displayed then extracts the menu item data and parses data into a string for each menu item for logging
+    if( [displayedMenuInteractions count] == 2 )
```

Comment:
Might want to add more comments to describe what is going on in this if-else. It's not very clear right now.